manifests/package.pp

```diff
@@ -3,32 +3,45 @@
 #
 # Install required packages
 #
-class caddy::package inherits caddy {
+class caddy::package (
+  $arch                  = $caddy::arch,
+  $install_path          = $caddy::install_path,
+  $caddy_tmp_dir         = $caddy::caddy_tmp_dir,
+  $caddy_license         = $caddy::caddy_license,
+  $caddy_account_id      = $caddy::caddy_account_id,
+  $caddy_api_key         = $caddy::caddy_api_key,
+  $caddy_telemetry       = $caddy::caddy_telemetry,
+  $caddy_features        = $caddy::caddy_features,
+  $caddy_private_devices = $caddy::caddy_private_devices,
+  $caddy_limit_processes = $caddy::caddy_limit_processes,
+) {
+
+  assert_private()
 
   Exec {
     path    => '/usr/bin:/usr/sbin:/bin:/usr/local/bin',
   }
 
   $caddy_url        = 'https://caddyserver.com/download/linux'
-  $caddy_dl_url     = "${caddy_url}/${caddy::params::arch}?plugins=${caddy::caddy_features}&license=${caddy::caddy_license}&telemetry=${caddy::caddy_telemetry}"
-  $caddy_dl_dir     = "${caddy::params::caddy_tmp_dir}/caddy_linux_${$caddy::params::arch}_custom.tar.gz"
-  $caddy_dl_command = $caddy::params::caddy_license ? {
+  $caddy_dl_url     = "${caddy_url}/${arch}?plugins=${caddy_features}&license=${caddy_license}&telemetry=${caddy_telemetry}"
+  $caddy_dl_dir     = "${caddy_tmp_dir}/caddy_linux_${$arch}_custom.tar.gz"
+  $caddy_dl_command = $caddy_license ? {
     'personal'   => "curl -o ${caddy_dl_dir} '${caddy_dl_url}'",
-    'commercial' => "curl -o ${caddy_dl_dir} '${caddy_dl_url}' --user ${caddy::params::caddy_account_id}:${caddy::params::caddy_api_key}"
+    'commercial' => "curl -o ${caddy_dl_dir} '${caddy_dl_url}' --user ${caddy_account_id}:${caddy_api_key}"
```

Comment:
this should use archive, but that's a separate issue.

Reply:
Exactly. I already created #30 for that, but wanted to do small pull requests.